sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_malloc_properties.asciidoc

```diff
@@ -0,0 +1,1319 @@
+
+[source,c++]
+----
+using namespace sycl::ext::oneapi::experimental;
+
+properties P4{bar, foo{1}};
+
+// APtr6 is of type annotated_ptr<int, decltype(properties{bar})>
+auto APtr6 = malloc_annotated<int>(N, q, sycl::usm::alloc::device, P4);
+
+// APtr7 is of type annotated_ptr<int, decltype(properties{})>;
+auto APtr7 = malloc_annotated<int>(N, q, sycl::usm::alloc::device);
+----
+
+If the USM memory allocation kind specified by a parameter to the allocation function is different than the USM memory allocation kind specified by the `sycl::ext::oneapi::experimental::usm_kind` property, the function throws a
+synchronous exception with the `errc::invalid` error code. 
+If the `sycl::ext::oneapi::experimental::usm_kind` property specifies a different USM memory allocation kind than the function supports, the compiler will issue a diagnostic error.
+
+[source,c++]
+----
+using namespace sycl::ext::oneapi::experimental;
+
+properties P5{usm_kind<sycl::usm::alloc::device>};
+
+// Throws an exception with error code errc::invalid
+auto APtr8 = malloc_annotated<int>(N, q, sycl::usm::alloc::host, P5);
+
+// Error: the USM kinds do not agree
+auto APtr9 = malloc_host_annotated<int>(N, q, P5);
+----
+
+The following example uses the compile-time-constant property `alignment`, defined in the link:../proposed/sycl_ext_oneapi_annotated_ptr.asciidoc[sycl_ext_oneapi_annotated_ptr] extension.
+When `alignment` is passed to a USM memory allocation function with `properties` support, it will appear on the returned `annotated_ptr` since it is a compile-time constant property.
+It also informs the runtime to allocate the memory with this alignment.
+ 
+[source,c++]
+----
+using namespace sycl::ext::oneapi::experimental;
+
+properties P7{alignment<512>};
+properties P8{alignment<2>};
+properties P9{alignment<64>};
+
+// APtr10 is of type annotated_ptr<int, decltype(properties{alignment<512>, usm_kind<sycl::usm::alloc::device>})>
+// The raw pointer of APtr10 is 512-byte aligned
+auto APtr10 = malloc_device_annotated<int>(N, q, P7);
+
+// APtr11 is of type annotated_ptr<int, decltype(properties{alignment<1>, usm_kind<sycl::usm::alloc::device>})>
+// The raw pointer of APtr11 is sizeof(int)-byte aligned, e.g., 4 for some implementations
+auto APtr11 = malloc_device_annotated<int>(N, q, P8);
+
+// APtr12 is of type annotated_ptr<void, decltype(properties{alignment<64>, usm_kind<sycl::usm::alloc::device>})>
+// The raw pointer of APtr12 is 512-byte aligned
+auto APtr12 = malloc_device_annotated(512, q, P9);
+----
+
+This extension also introduces USM memory allocation functions with `properties` support that allow alignment to be specified at runtime, using a separate parameter of type `size_t`.
+If the parameter of type `size_t` has value `A` and the compile-time constant `alignment` property has value `B`, the resulting pointer will be both `A`-byte aligned and `B`-byte aligned, as well as `C`-byte aligned,
+where `C` is the least common multiple of `A` and `B`.
```

Comment:
It would be good to note here that the returned `annotated_ptr` will not have the `alignment` property because the alignment is a runtime quantity.

Reply:
Done.